js/src/Ice/ConnectionI.js

```diff
@@ -915,13 +915,13 @@ export class ConnectionI {
                     }
                     this._batchRequestQueue.destroy(this._exception);
                     this._transceiver.unregister();
+                    this._transceiver.close();
```

Comment:
This matches what we do in other implementations, close the socket when we enter StateClosed, and destroy it when we enter StateFinished.